src/com/owncloud/android/utils/FileStorageUtils.java

```diff
@@ -154,7 +154,11 @@ public static String getParentPath(String remotePath) {
     public static OCFile fillOCFile(RemoteFile remote) {
         OCFile file = new OCFile(remote.getRemotePath());
         file.setCreationTimestamp(remote.getCreationTimestamp());
-        file.setFileLength(remote.getLength());
+        if (remote.getMimeType().equalsIgnoreCase("DIR")){
+            file.setFileLength(remote.getSize());
```

Comment:
Not sure, could be OC 6 or OC 7. That property has been there forever... But it's only on folders as far as I remember, because files have "getcontentlength" but folders don't have that one officially.